src/e2e-test/features/bigquery/source/BigQueryToGCS_WithMacro.feature

```diff
@@ -69,3 +69,223 @@ Feature: BigQuery source - Verification of BigQuery to GCS successful data trans
+    Then Enter runtime argument value "csvFormat" for key "gcsFormat"
+    Then Enter runtime argument value "cmekGCS" for GCS cmek property key "cmekGCS" if GCS cmek is enabled
+    Then Run the preview of pipeline with runtime arguments
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Click on preview data for GCS sink
+    Then Close the preview data
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "projectId" for key "bqProjectId"
+    Then Enter runtime argument value "projectId" for key "bqDatasetProjectId"
+    Then Enter runtime argument value "filter" for key "bqFilter"
+    Then Enter runtime argument value "serviceAccountType" for key "serviceAccountType"
+    Then Enter runtime argument value "serviceAccount" for key "serviceAccount"
+    Then Enter runtime argument value "dataset" for key "bqDataset"
+    Then Enter runtime argument value for BigQuery source table name key "bqSourceTable"
+    Then Enter runtime argument value "projectId" for key "gcsProjectId"
+    Then Enter runtime argument value for GCS sink property path key "gcsSinkPath"
+    Then Enter runtime argument value "gcsPathDateSuffix" for key "gcsPathSuffix"
+    Then Enter runtime argument value "csvFormat" for key "gcsFormat"
+    Then Enter runtime argument value "cmekGCS" for GCS cmek property key "cmekGCS" if GCS cmek is enabled
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Verify data is transferred to target GCS bucket
+    Then Validate the cmek key "cmekGCS" of target GCS bucket if cmek is enabled
+
+  @CMEK @BQ_SOURCE_TEST @GCS_SINK_TEST
+  Scenario:Validate successful records transfer from BigQuery to GCS with macro arguments for output schema
+    Given Open Datafusion Project to configure pipeline
+    When Source is BigQuery
+    When Sink is GCS
+    Then Open BigQuery source properties
+    Then Enter BigQuery property reference name
+    Then Enter BigQuery property "projectId" as macro argument "bqProjectId"
+    Then Enter BigQuery property "datasetProjectId" as macro argument "bqDatasetProjectId"
+    Then Enter BigQuery property "serviceAccountType" as macro argument "serviceAccountType"
+    Then Enter BigQuery property "serviceAccountFilePath" as macro argument "serviceAccount"
+    Then Enter BigQuery property "serviceAccountJSON" as macro argument "serviceAccount"
+    Then Enter BigQuery property "dataset" as macro argument "bqDataset"
+    Then Enter BigQuery property "table" as macro argument "bqSourceTable"
+    Then Enter BigQuery source property output schema "outputSchema" as macro argument "bqOutputSchema"
```

Comment:
Use the existing step definition for selecting macro action for output schema as mentioned above.